### PR TITLE
Fix task api status code response for malformed id in the scheduler

### LINKED ISSUE #2943: Download task details with wrong ID throws 500 error instead of 404
To reproduce
- Go to the task list
- reschedule task
- expand the exapndo row and copy link of "Download task data" 
- Open link in a new browser
- Change the id for something else
- See 500 error Scheduler

```
2024-05-13 15:58:46 [SQL: SELECT tasks.id AS tasks_id, tasks.scheduler_id AS tasks_scheduler_id, tasks.type AS tasks_type, tasks.p_item AS tasks_p_item, tasks.status AS tasks_status, tasks.created_at AS tasks_created_at, tasks.modified_at AS tasks_modified_at 
2024-05-13 15:58:46 FROM tasks 
2024-05-13 15:58:46 WHERE tasks.id = %(id_1)s::UUID 
2024-05-13 15:58:46  LIMIT %(param_1)s]
2024-05-13 15:58:46 [parameters: {'id_1': 'ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80', 'param_1': 1}]
2024-05-13 15:58:46 (Background on this error at: https://sqlalche.me/e/20/9h9h)
2024-05-13 15:58:46 Traceback (most recent call last):
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1960, in _exec_single_context
2024-05-13 15:58:46     self.dialect.do_execute(
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 924, in do_execute
2024-05-13 15:58:46     cursor.execute(statement, parameters)
2024-05-13 15:58:46 psycopg2.errors.InvalidTextRepresentation: invalid input syntax for type uuid: "ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80"
2024-05-13 15:58:46 LINE 3: WHERE tasks.id = 'ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80'::UUID...
2024-05-13 15:58:46                          ^
2024-05-13 15:58:46 
2024-05-13 15:58:46 
2024-05-13 15:58:46 The above exception was the direct cause of the following exception:
2024-05-13 15:58:46 
2024-05-13 15:58:46 Traceback (most recent call last):
2024-05-13 15:58:46   File "/app/scheduler/scheduler/server/server.py", line 419, in get_task
2024-05-13 15:58:46     task = self.ctx.datastores.task_store.get_task_by_id(task_id)
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/app/scheduler/scheduler/storage/storage.py", line 48, in wrapper
2024-05-13 15:58:46     return func(*args, **kwargs)
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/app/scheduler/scheduler/storage/task_store.py", line 63, in get_task_by_id
2024-05-13 15:58:46     task_orm = session.query(models.TaskDB).filter(models.TaskDB.id == task_id).first()
2024-05-13 15:58:46                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/query.py", line 2727, in first
2024-05-13 15:58:46     return self.limit(1)._iter().first()  # type: ignore
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/query.py", line 2826, in _iter
2024-05-13 15:58:46     result: Union[ScalarResult[_T], Result[_T]] = self.session.execute(
2024-05-13 15:58:46                                                   ^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2306, in execute
2024-05-13 15:58:46     return self._execute_internal(
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2191, in _execute_internal
2024-05-13 15:58:46     result: Result[Any] = compile_state_cls.orm_execute_statement(
2024-05-13 15:58:46                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/context.py", line 293, in orm_execute_statement
2024-05-13 15:58:46     result = conn.execute(
2024-05-13 15:58:46              ^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1408, in execute
2024-05-13 15:58:46     return meth(
2024-05-13 15:58:46            ^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/sql/elements.py", line 513, in _execute_on_connection
2024-05-13 15:58:46     return connection._execute_clauseelement(
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1630, in _execute_clauseelement
2024-05-13 15:58:46     ret = self._execute_context(
2024-05-13 15:58:46           ^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1839, in _execute_context
2024-05-13 15:58:46     return self._exec_single_context(
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1979, in _exec_single_context
2024-05-13 15:58:46     self._handle_dbapi_exception(
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 2335, in _handle_dbapi_exception
2024-05-13 15:58:46     raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1960, in _exec_single_context
2024-05-13 15:58:46     self.dialect.do_execute(
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 924, in do_execute
2024-05-13 15:58:46     cursor.execute(statement, parameters)
2024-05-13 15:58:46 sqlalchemy.exc.DataError: (psycopg2.errors.InvalidTextRepresentation) invalid input syntax for type uuid: "ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80"
2024-05-13 15:58:46 LINE 3: WHERE tasks.id = 'ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80'::UUID...
2024-05-13 15:58:46                          ^
2024-05-13 15:58:46 
2024-05-13 15:58:46 [SQL: SELECT tasks.id AS tasks_id, tasks.scheduler_id AS tasks_scheduler_id, tasks.type AS tasks_type, tasks.p_item AS tasks_p_item, tasks.status AS tasks_status, tasks.created_at AS tasks_created_at, tasks.modified_at AS tasks_modified_at 
2024-05-13 15:58:46 FROM tasks 
2024-05-13 15:58:46 WHERE tasks.id = %(id_1)s::UUID 
2024-05-13 15:58:46  LIMIT %(param_1)s]
2024-05-13 15:58:46 [parameters: {'id_1': 'ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80', 'param_1': 1}]
2024-05-13 15:58:46 (Background on this error at: https://sqlalche.me/e/20/9h9h)
2024-05-13 15:58:46 Traceback (most recent call last):
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1960, in _exec_single_context
2024-05-13 15:58:46     self.dialect.do_execute(
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 924, in do_execute
2024-05-13 15:58:46     cursor.execute(statement, parameters)
2024-05-13 15:58:46 psycopg2.errors.InvalidTextRepresentation: invalid input syntax for type uuid: "ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80"
2024-05-13 15:58:46 LINE 3: WHERE tasks.id = 'ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80'::UUID...
2024-05-13 15:58:46                          ^
2024-05-13 15:58:46 
2024-05-13 15:58:46 
2024-05-13 15:58:46 The above exception was the direct cause of the following exception:
2024-05-13 15:58:46 
2024-05-13 15:58:46 Traceback (most recent call last):
2024-05-13 15:58:46   File "/app/scheduler/scheduler/server/server.py", line 419, in get_task
2024-05-13 15:58:46     task = self.ctx.datastores.task_store.get_task_by_id(task_id)
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/app/scheduler/scheduler/storage/storage.py", line 48, in wrapper
2024-05-13 15:58:46     return func(*args, **kwargs)
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/app/scheduler/scheduler/storage/task_store.py", line 63, in get_task_by_id
2024-05-13 15:58:46     task_orm = session.query(models.TaskDB).filter(models.TaskDB.id == task_id).first()
2024-05-13 15:58:46                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/query.py", line 2727, in first
2024-05-13 15:58:46     return self.limit(1)._iter().first()  # type: ignore
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/query.py", line 2826, in _iter
2024-05-13 15:58:46     result: Union[ScalarResult[_T], Result[_T]] = self.session.execute(
2024-05-13 15:58:46                                                   ^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2306, in execute
2024-05-13 15:58:46     return self._execute_internal(
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 2191, in _execute_internal
2024-05-13 15:58:46     result: Result[Any] = compile_state_cls.orm_execute_statement(
2024-05-13 15:58:46                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/orm/context.py", line 293, in orm_execute_statement
2024-05-13 15:58:46     result = conn.execute(
2024-05-13 15:58:46              ^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1408, in execute
2024-05-13 15:58:46     return meth(
2024-05-13 15:58:46            ^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/sql/elements.py", line 513, in _execute_on_connection
2024-05-13 15:58:46     return connection._execute_clauseelement(
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1630, in _execute_clauseelement
2024-05-13 15:58:46     ret = self._execute_context(
2024-05-13 15:58:46           ^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1839, in _execute_context
2024-05-13 15:58:46     return self._exec_single_context(
2024-05-13 15:58:46            ^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1979, in _exec_single_context
2024-05-13 15:58:46     self._handle_dbapi_exception(
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 2335, in _handle_dbapi_exception
2024-05-13 15:58:46     raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 1960, in _exec_single_context
2024-05-13 15:58:46     self.dialect.do_execute(
2024-05-13 15:58:46   File "/usr/local/lib/python3.11/site-packages/sqlalchemy/engine/default.py", line 924, in do_execute
2024-05-13 15:58:46     cursor.execute(statement, parameters)
2024-05-13 15:58:46 sqlalchemy.exc.DataError: (psycopg2.errors.InvalidTextRepresentation) invalid input syntax for type uuid: "ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80"
2024-05-13 15:58:46 LINE 3: WHERE tasks.id = 'ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80'::UUID...
2024-05-13 15:58:46                          ^
2024-05-13 15:58:46 
2024-05-13 15:58:46 [SQL: SELECT tasks.id AS tasks_id, tasks.scheduler_id AS tasks_scheduler_id, tasks.type AS tasks_type, tasks.p_item AS tasks_p_item, tasks.status AS tasks_status, tasks.created_at AS tasks_created_at, tasks.modified_at AS tasks_modified_at 
2024-05-13 15:58:46 FROM tasks 
2024-05-13 15:58:46 WHERE tasks.id = %(id_1)s::UUID 
2024-05-13 15:58:46  LIMIT %(param_1)s]
2024-05-13 15:58:46 [parameters: {'id_1': 'ea59f5d0-c6ab-42f8-a40e-d1bbd72dc80', 'param_1': 1}]
2024-05-13 15:58:46 (Background on this error at: https://sqlalche.me/e/20/9h9h)
```

### PR DESCRIPTION
**Pull request by jpbruinsslot**
### Changes

A 500 was returned on malformed id's when issuing requests to the `/tasks/{id}` endpoint. This PR solves this by correctly relaying a 400 status code error. I opted for catching exceptions from the `storage` module by implementing a `StorageError` exception, in order to keep the server API clean and not litter it with specific sqlalchemy, and psycopg2 exception handling.

### Issue link

Closes  #2943

### CODE REVIEW

**Comment by ammar92** [comment]
User requests with malformed input should indeed yield a `400` status response. However, I'm still a bit hesitant about throwing `400`s for every `StorageException`, since errors such as `External service returned an error.` doesn't necessarily indicate that the request is wrong. Perhaps within this context it's mostly a client error (due to a parse error to UUIDs?), but the error message is misleading.

I was also wondering; when should it be a not found error or a malformed id error? They're very identical, because with an invalid format you could still say that the entity cannot be found since it simply doesn't exit. But, if `task_id` is, for example, an expected UUID, then that could be validated. FastAPI supports a few extra data types, including UUIDs, that can be validated, even on on path operations. E.g.:
```python3
@app.put("/items/{item_id}")
def read_items(
    item_id: UUID,
    ...
):
```

Let me know what you think. I do like the framework you implemented to handle these errors.

**Comment by jpbruinsslot** [comment]
I agree with the points that you've raised. I didn't realize that fastapi could leverage the fastapi validation for it that would make it more elegant.

**Comment by jpbruinsslot** [comment]
@ammar92 I've updated the pr and add the fastapi uuid suggestion. I've restructured the `StorageError` a bit such that it is still a 500 when an exception occurs within the `storage` module. Additionally I updated the `FilterError` in the `filters` module, to make sure that those are specifically handled instead of a generic `StorageError`

**Review by ammar92 — APPROVED** [review]
Looks good to me 👍

**Comment by jpbruinsslot** [comment]
Just fixing some pre-commit issues

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [ ] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [ ] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
- Onboarding flow works
- (Re)Scheduling of tasks work.
- Findings are created

### What doesn't work:
When reproducing the steps as described in #2943 I get the following error: 

image


```
rocky-1  | [21/May/2024 09:15:41] "GET /en/aa/tasks/d42ccb45-8aa5-4d5d-bfbe-02f295c90df0/download/ HTTP/1.1" 200 494
rocky-1  | Internal Server Error: /en/aa/tasks/d42ccb45-8aa5-4d5d-bfbe-02f295c90ueue/download/
rocky-1  | Traceback (most recent call last):
rocky-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
rocky-1  |     response = get_response(request)
rocky-1  |                ^^^^^^^^^^^^^^^^^^^^^
rocky-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
rocky-1  |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
rocky-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
rocky-1  |   File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
rocky-1  |     return self.dispatch(request, *args, **kwargs)
rocky-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
rocky-1  |   File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 143, in dispatch
rocky-1  |     return handler(request, *args, **kwargs)
rocky-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
rocky-1  |   File "/app/rocky/rocky/views/tasks.py", line 28, in get
rocky-1  |     task_details = client.get_task_details(self.organization.code, task_id)
rocky-1  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
rocky-1  |   File "/app/rocky/rocky/scheduler.py", line 227, in get_task_details
rocky-1  |     res.raise_for_status()
rocky-1  |   File "/usr/local/lib/python3.11/site-packages/httpx/_models.py", line 761, in raise_for_status
rocky-1  |     raise HTTPStatusError(message, request=request, response=self)
rocky-1  | httpx.HTTPStatusError: Client error '422 Unprocessable Entity' for url 'http://scheduler:8000/tasks/d42ccb45-8aa5-4d5d-bfbe-02f295c90ueue'
rocky-1  | For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422
rocky-1  | Internal Server Error: /en/aa/tasks/d42ccb45-8aa5-4d5d-bfbe-02f295c90ueue/download/
rocky-1  | Traceback (most recent call last):
rocky-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
rocky-1  |     response = get_response(request)
rocky-1  |                ^^^^^^^^^^^^^^^^^^^^^
rocky-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
rocky-1  |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
rocky-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
rocky-1  |   File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
rocky-1  |     return self.dispatch(request, *args, **kwargs)
rocky-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
rocky-1  |   File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 143, in dispatch
rocky-1  |     return handler(request, *args, **kwargs)
rocky-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
rocky-1  |   File "/app/rocky/rocky/views/tasks.py", line 28, in get
rocky-1  |     task_details = client.get_task_details(self.organization.code, task_id)
rocky-1  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
rocky-1  |   File "/app/rocky/rocky/scheduler.py", line 227, in get_task_details
rocky-1  |     res.raise_for_status()
rocky-1  |   File "/usr/local/lib/python3.11/site-packages/httpx/_models.py", line 761, in raise_for_status
rocky-1  |     raise HTTPStatusError(message, request=request, response=self)
rocky-1  | httpx.HTTPStatusError: Client error '422 Unprocessable Entity' for url 'http://scheduler:8000/tasks/d42ccb45-8aa5-4d5d-bfbe-02f295c90ueue'
rocky-1  | For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422
rocky-1  | [21/May/2024 09:15:47] "GET /en/aa/tasks/d42ccb45-8aa5-4d5d-bfbe-02f295c90
```

### Bug or feature?:
See above.

File: mula/scheduler/storage/errors.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import functools
+
+import sqlalchemy
+
+
+class StorageError(Exception):
+    pass
+
+
+def exception_handler(func):
```
Comment (Donnype) [inline mula/scheduler/storage/errors.py:10]:
```suggestion
def db_exception_handler(func):
```
?

File: mula/scheduler/server/server.py [hunk 2]
```diff
@@ -398,12 +402,12 @@ def list_tasks(
...
-                detail=str(exc),
+                detail=f"invalid filter(s) [exception: {exc}]",
             ) from exc
-        except ValueError as exc:
+        except storage.errors.StorageError as exc:
             raise fastapi.HTTPException(
-                status_code=fastapi.status.HTTP_400_BAD_REQUEST,
-                detail=str(exc),
+                status_code=fastapi.status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail=f"error occurred while accessing the database [exception: {exc}]",
```
Comment (Donnype) [inline mula/scheduler/server/server.py:410]: We could consider a generic `@app.exception_handler` as well, which would clean up the router code but forces you to combing thinking about Exception hierarchy with a hierarchy of exception_handlers.

File: mula/tests/integration/test_api.py [hunk 3]
```diff
@@ -505,6 +595,16 @@ def test_get_task(self):
         self.assertEqual(200, response_get.status_code, 200)
         self.assertEqual(initial_item_id, response_get.json().get("id"))
 
+    def test_get_task_malformed_id(self):
+        response = self.client.get("/tasks/123.123")
+        self.assertEqual(422, response.status_code)
```
Comment (Donnype) [inline mula/tests/integration/test_api.py:600]: :+1:

**Review by Donnype — APPROVED** [review]
:ok_hand: 

Check out the suggestions!

**Comment by Rieven** [comment]
> ## Checklist for QA:
> * [x]  I have checked out this branch, and successfully ran a fresh `make reset`.
> * [ ]  I confirmed that there are no unintended functional regressions in this branch:
>   
>   * [x]  I have managed to pass the onboarding flow
>   * [x]  Objects and Findings are created properly
>   * [x]  Tasks are created and completed properly
> * [ ]  I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
> * [ ]  I checked the logs for errors and/or warnings and made issues where necessary
> 
> ### What works:
> * Onboarding flow works
> * (Re)Scheduling of tasks work.
> * Findings are created
> 
> ### What doesn't work:
> When reproducing the steps as described in #2943 I get the following error:
> 
> image

@stephanie0x00 This will be solved in this PR: https://github.com/minvws/nl-kat-coordination/pull/1957

File: mula/scheduler/server/server.py (hunk 2 above)
Comment (jpbruinsslot) [reply mula/scheduler/server/server.py:410]: Yes I agree, let's pick it up in a new issue

**Comment by jpbruinsslot** [comment]
For QA: since it is obscure to trigger this error, the 500 error is resolved in the scheduler, and a PR to handle the 422 status error code is being finalized, I would suggest to merge this so that the new rocky PR has the updated status code reporting from the scheduler.

**Review by underdarknl — APPROVED** [review]

**Pull request merged**